### PR TITLE
Reorganize components user guide into other doc sections

### PR DESCRIPTION
**Pull request by droumis**
Addresses a small part of #3869 (also see #4251) by reorganizing the Components User Guide, and in the process initializes the Background section.

image

How To > Components:

image

image

### CODE REVIEW

**Comment by philippjfr** [comment]
Looking great, can you:

- Add `Background` to the top-level TOC
- Remove doc/user_guide/Components.md (as long as all contents have now been migrated)

File: doc/background/components/components_overview.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
+# Component overview
+
+Panel provides a wide range of components for easily composing panels, apps, and dashboards both in the notebook and as standalone apps. The components can be broken down into three broad classes of objects:
+
+* ``Pane`` objects allow wrapping external viewable items like Bokeh, Plotly, Vega, or HoloViews plots, so they can be embedded in a panel.
+* ``Widget`` objects provide controls that can trigger Python or JavaScript events.
+* ``Panel`` layout objects allow combining plots into a ``Row``, ``Column``, ``Tabs`` or a ``Grid``.
+
+All objects share an API that makes it easy to [customize their layout behavior and visual appearance](./Customization.md), [link](./Links.md) and [display and export](./Display_and_Export.rst) them. To display any panel objects in a notebook environment ensure you load the extension first:
```
Comment (philippjfr) [inline doc/background/components/components_overview.md:9]: These cross-references are invalid and should link to how-to sections OR Background stubs.

File: doc/background/components/components_overview.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+```{pyodide}
+import panel as pn
+pn.extension()
+```
+
+Note that to use certain components such as Vega, LaTeX, and Plotly plots in a notebook, the models must be loaded using the extension. E.g.:
+
+    pn.extension('vega', 'katex')
+
+will ensure that the Vega and LaTeX JS dependencies are loaded. Once the extension is loaded, Panel objects will display themselves in the notebook, outside the notebook objects can be displayed in a server using the show method or run from the commandline by appending ``.serveable()`` to the objects to be displayed.
```
Comment (philippjfr) [inline doc/background/components/components_overview.md:21]:
```suggestion
will ensure that the Vega and LaTeX JS dependencies are loaded. Once the extension is loaded, Panel objects will display themselves in the notebook, outside the notebook objects can be displayed in a server using the show method or run from the commandline by appending ``.servable()`` to the objects to be displayed.
```

File: doc/background/index.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+:::{grid-item-card} {octicon}`multi-select;2.5em;sd-mr-1` Param
+:link: param/param
+:link-type: doc
+
+Discover why Panel utilizes the Param library.
+:::
+
+:::{grid-item-card} {octicon}`git-compare;2.5em;sd-mr-1` Library Comparisons
+:link: ../about/comparisons
```
Comment (philippjfr) [inline doc/background/index.md:32]: Should we move that section from about -> background?

File: doc/how_to/components/construct_panes.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+Alternatively, you can create a pane using the `pn.panel()` utility to automatically infer the pane type from the object being passed as the argument:
+
+```{pyodide}
+png = pn.panel('https://upload.wikimedia.org/wikipedia/commons/4/47/PNG_transparency_demonstration_1.png', width=500)
+
+png
+```
+
+:::{admonition} See Also
+:class: seealso
```
Comment (philippjfr) [inline doc/how_to/components/construct_panes.md:27]: Is this a valid class?

File: doc/how_to/components/widget_values.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# Access and Set Widget Values
+
+In addition to other parameters that govern widget behavior and appearance, Widget objects have a ``value`` parameter that can be used to access the current value state.
+
+
+Let's first create a TextInput widget:
```
Comment (philippjfr) [inline doc/how_to/components/widget_values.md:6]:
```suggestion
Let's first create a `TextInput` widget:
```

File: doc/how_to/components/construct_panes.md (hunk 4 above)
Comment (droumis) [reply doc/how_to/components/construct_panes.md:27]: [According to MyST](https://myst.tools/docs/spec/admonitions#:~:text=hint%22%20%7C%20%22important%22%20%7C%20%22note%22%20%7C%20%22-,seealso,-%22%20%7C%20%22tip%22%20%7C%20%22warning%22)
), yes. Although it appears that you are flattening the styling of anything but {important}, is that correct? Sorry, I'm not fully understanding this system yet.

File: doc/index.md [hunk 6]
```diff
@@ -157,6 +157,7 @@ self
 getting_started/index
 how_to/index
 gallery/index
+background/index
```
Comment (droumis) [inline doc/index.md:160]: > * Add `Background` to the top-level TOC


I think it already is, no?

@philippjfr, are you referring to something else?

Comment (philippjfr) [reply doc/index.md:160]: Missed it sorry!

File: doc/background/index.md (hunk 3 above)
Comment (philippjfr) [reply doc/background/index.md:32]: I'm not fully convinced we need an About page tbh.

File: doc/background/components/components_overview.md (hunk 1 above)
Comment (droumis) [reply doc/background/components/components_overview.md]: done

**Review by philippjfr — APPROVED** [review]
Thanks for the fixes! Looking good.

**Comment by philippjfr** [comment]
Did you want to remove Remove doc/user_guide/Components.md?

File: doc/background/index.md (hunk 3 above)
Comment (droumis) [reply doc/background/index.md]: ok, we can save the extermination of About for another day, but I've moved comparisons to background

**Comment by droumis** [comment]
> Did you want to remove Remove doc/user_guide/Components.md?

Yep! just removed it. Should be ready to merge now if those were all the comments

**Comment by philippjfr** [comment]
Perfect, will just go ahead and merge without waiting on tests since those changes shouldn't affect anything.

**Pull request merged**